src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -1038,8 +1039,11 @@ public static boolean onEffectRemoved(LivingEntity entity, MobEffectInstance eff
     public static int getEnchantmentLevelSpecific(int level, ItemStack stack, Holder<Enchantment> ench) {
         var enchantments = new ItemEnchantments.Mutable(ItemEnchantments.EMPTY);
         enchantments.set(ench, level);
-        var event = new GetEnchantmentLevelEvent(stack, enchantments, ench);
-        NeoForge.EVENT_BUS.post(event);
+        RegistryLookup<Enchantment> lookup = ench.unwrapLookup();
+        if (lookup != null) { // Pretty sure this is never null, but I can't *prove* that it isn't.
```

Comment:
Can we restructure the body of this event to avoid the extra instantiation in case lookup does end up being null? For example:
```java
RegistryLookup<Enchantment> lookup = ench.unwrapLookup();
if (lookup == null) { // Pretty sure this is never null, but I can't *prove* that it isn't.
    return level;
}
var enchantments = new ItemEnchantments.Mutable(ItemEnchantments.EMPTY);
enchantments.set(ench, level);
var event = new GetEnchantmentLevelEvent(stack, enchantments, ench, lookup);
NeoForge.EVENT_BUS.post(event);
return enchantments.getLevel(ench);
```